### ISSUE TITLE
bit.arshift treats all inputs as 0x80000000 if bit 31 is set to 1

### ISSUE THREAD

**Issue by lunjesse** [TASVideos/BizHawk#1825]
This was tested on BizHawk 2.3.1 to 2.4.

Steps:
1. Open the lua console.
2. Type bit.arshift(0x80000000, 1), where any hex digit can replace the 0's after 8
3. Press enter
The results would always be -1073741824, which is the same as bit.arshift(0x80000000, 1). Eg. The following all give the same result of -1073741824 (0xC0000000):
bit.arshift(0x8ABCDE00, 1)   -  Should be 0xC55E6F00
bit.arshift(0x80000010, 1) - Should be 0xC0000008
bit.arshift(0xF0000000, 1) - Should be 0xF8000000

In other words, once bit 31 is set, bit.arshift completely ignores all other bits, and treats the input as 0x80000000. 

The examples uses a shift of 1, but this applies to different shift amounts as well.

**Comment by zeromus** [comment]
well, lua sucks, and bizhawk's lua is subject to even more of the basic numerical type suckiness that afflicts lua, so you're nuts to expect any sensible behavior here, but okay, it seems useful for emulation and I saw a bug so I fixed it.
